soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -4620,6 +4620,85 @@ void CreateScrubMessages() {
     }
 }
 
+void CreateRupeeMessages() {
+    CustomMessageManager* customMessageManager = CustomMessageManager::Instance;
+    customMessageManager->AddCustomMessageTable(Randomizer::rupeeMessageTableID);
+    const std::vector<u8> rupees = { TEXT_BLUE_RUPEE, TEXT_RED_RUPEE, TEXT_PURPLE_RUPEE, TEXT_HUGE_RUPEE };
+    std::string rupeeText = " ";
+    for (u8 rupee : rupees) {
+        switch (rupee) {
+            case TEXT_BLUE_RUPEE:
+                rupeeText = "\x05\x03 5 %RUPEE%\x05\x00";
+                break;
+            case TEXT_RED_RUPEE:
+                rupeeText = "\x05\x01 20 %RUPEE%\x05\x00";
+                break;
+            case TEXT_PURPLE_RUPEE:
+                rupeeText = "\x05\x05 50 %RUPEE%\x05\x00";
+                break;
+            case TEXT_HUGE_RUPEE:
+                rupeeText = "\x05\x06 200 %RUPEE%\x05\x00";
+                break;
+        }
+        customMessageManager->CreateMessage(Randomizer::rupeeMessageTableID, rupee,
+            { TEXTBOX_TYPE_BLACK, TEXTBOX_POS_BOTTOM,
+                "You found" + rupeeText + " !",
+                "Du hast" + rupeeText + "  gefunden!",
+                "Vous obtenez" + rupeeText + " !"
+        });
+    }
+}
+
+std::string Randomizer::Randomizer_InsertRupeeName(std::string message, int language) {
```

Comment:
I think I had issues with passing through a CustomMessageEntry because this code was located somewhere else. I'm not entirely sure anymore, I did a decent bit of struggling before I got something that worked.

`CustomMessageManager::CreateMessage` and `CustomMessageManager::CreateGetItemMessage` do a similar thing, so I might've taken some uhh.. inspiration there as well.

Anyway, I can change it to a loop inside the function instead if you'd prefer that (and move it all inside `Randomizer_GetRupeeMessage()`), just lemme know.

Reply:
My issue was more so that by calling this function individually for each string and each language, it's reinstantiating the arrays and allocating more memory for them each time. Taking in a CustomMessageEntry instead would at least make it so they only get instantiated once per entry instead of 3 times. Although thinking about it more, it may make more sense to declare these arrays outside of this function. They could also probably be static as well, and that would resolve that issue nicely.